data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/producer/KafkaCustomProducer.java

```diff
@@ -183,7 +184,8 @@ public boolean validateSchema(final String jsonData, final String schemaJson) th
     private Callback callBack(final Object dataForDlq) {
         return (metadata, exception) -> {
             if (null != exception) {
-                LOG.error("Error occured while publishing " + exception.getMessage());
+                LOG.error("Error occurred while publishing " + exception.getMessage());
```

Comment:
Please update this line to use SLF4J's string interpolation:

```
LOG.error("Error occurred while publishing ", exception.getMessage());
```

Reply:
Fixed